### ISSUE TITLE
Parallelize Local Regression

### ISSUE THREAD

**Issue by ryanbanderson** [USGS-Astrogeology/PyHAT_Point_Spectra_GUI#283]
This method is a prime candidate for parallel processing.

**Comment by tisaconundrum2** [comment]
Multiprocessing is great for CPU intensive situations. 
Here's an example function that could be used. 

    def multiprocessing(func, args, workers):
        with ProcessPoolExecutor(work) as ex:
        	  res = ex.map(func, args)
        return list(res)

**Comment by ryanbanderson** [comment]
Yeah, I've been learning more about multiprocessing. What I'm not sure on is if we can use it within our existing code structure. A lot of the examples online assume you're working from __main__ directly. Not much about including multiprocessing as part of a class.

**Comment by tisaconundrum2** [comment]
Could do something like this and see what happens?

    def jog(self):
        method = self.chooseAlgorithmComboBox.currentText()
        xvars = [str(x.text()) for x in self.xVariableList.selectedItems()]
        yvars = [('comp', str(y.text())) for y in self.yVariableList.selectedItems()]
        fit_intercept = self.fit_intercept.isChecked()
        force_positive = self.forcepositive.isChecked()
        params = {'fit_intercept': self.fit_intercept.isChecked(),
                  'max_iter': 10000,
                  'positive': self.forcepositive.isChecked(),
                  'selection': 'random'}
        localmodel = local_regression.LocalRegression(params, n_neighbors = self.n_neighbors_spin.value())
        traindata = self.data[self.choosedata_train.currentText()]
        predictdata = self.data[self.choosedata_predict.currentText()]
        x_train = np.array(traindata.df[xvars])
        y_train = np.array(traindata.df[yvars])
        x_predict = np.array(predictdata.df[xvars])
        predictions, coefs, intercepts = localmodel.fit_predict(x_train, y_train, x_predict)
        predictname = ('predict', 'Local LASSO - ' + self.choosedata_predict.currentText() + ' - Predict')
        self.data[self.choosedata_predict.currentText()].df[predictname] = predictions


        coefs = pd.DataFrame(coefs, columns = pd.MultiIndex.from_tuples(self.data[self.choosedata_predict.currentText()].df[xvars].columns.values))
        coefs[('meta', 'Intercept')] = intercepts
        try:
            self.data['Model Coefficients'] = spectral_data(pd.concat([self.data['Model Coefficients'].df, coefs]))
        except:
            self.data['Model Coefficients'] = spectral_data(coefs)
            self.datakeys.append('Model Coefficients')

    def multiprocessing(func, args, workers):
        with ProcessPoolExecutor(work) as ex:
        	  res = ex.map(func, args)
        return list(res)

    def run(self):
        multiprocessing(jog, self, 3)
